examples/hacker_news/hacker_news/jobs/download_job.py

```diff
@@ -0,0 +1,104 @@
+SNOWFLAKE_CONF = {
+    "account": os.getenv("SNOWFLAKE_ACCOUNT", ""),
+    "user": os.getenv("SNOWFLAKE_USER", ""),
+    "password": os.getenv("SNOWFLAKE_PASSWORD", ""),
+    "database": "DEMO_DB",
+    "warehouse": "TINY_WAREHOUSE",
+}
+
+# the configuration we'll need to make spark able to read from / write to s3
+configured_pyspark = pyspark_resource.configured(
+    {
+        "spark_conf": {
+            "spark.jars.packages": "com.amazonaws:aws-java-sdk:1.7.4,org.apache.hadoop:hadoop-aws:2.7.7",
+            "spark.hadoop.fs.s3.impl": "org.apache.hadoop.fs.s3native.NativeS3FileSystem",
+            "spark.hadoop.fs.s3.awsAccessKeyId": os.getenv("AWS_ACCESS_KEY_ID", ""),
+            "spark.hadoop.fs.s3.awsSecretAccessKey": os.getenv("AWS_SECRET_ACCESS_KEY", ""),
+            "spark.hadoop.fs.s3.buffer.dir": "/tmp",
+        }
+    }
+)
+
+DOWNLOAD_RESOURCES_STAGING = {
+    "io_manager": s3_pickle_io_manager.configured({"s3_bucket": "hackernews-elementl-dev"}),
+    "s3": s3_resource,
+    "partition_start": ResourceDefinition.string_resource(),
+    "partition_end": ResourceDefinition.string_resource(),
+    "parquet_io_manager": partitioned_parquet_io_manager.configured(
+        {"base_path": "s3://hackernews-elementl-dev"}
+    ),
+    "warehouse_io_manager": time_partitioned_snowflake_io_manager.configured(SNOWFLAKE_CONF),
+    "pyspark": configured_pyspark,
+    "hn_client": hn_api_subsample_client.configured({"sample_rate": 10}),
+    "base_url": ResourceDefinition.hardcoded_resource("http://demo.elementl.dev", "Dagit URL"),
+}
+
+DOWNLOAD_RESOURCES_PROD = {
+    "io_manager": s3_pickle_io_manager.configured({"s3_bucket": "hackernews-elementl-prod"}),
+    "s3": s3_resource,
+    "partition_start": ResourceDefinition.string_resource(),
+    "partition_end": ResourceDefinition.string_resource(),
+    "parquet_io_manager": partitioned_parquet_io_manager.configured(
+        {"base_path": "s3://hackernews-elementl-prod"}
+    ),
+    "warehouse_io_manager": time_partitioned_snowflake_io_manager.configured(SNOWFLAKE_CONF),
+    "pyspark": configured_pyspark,
+    "hn_client": hn_api_subsample_client.configured({"sample_rate": 10}),
+}
+
+
+DEFAULT_PARTITION_RESOURCE_CONFIG = {
+    "partition_start": {"config": "2020-12-30 00:00:00"},
+    "partition_end": {"config": "2020-12-30 01:00:00"},
+}
+
+DOWNLOAD_TAGS = {
+    "dagster-k8s/config": {
+        "container_config": {
+            "resources": {
+                "requests": {"cpu": "500m", "memory": "2Gi"},
+            }
+        },
+    }
+}
+
+
+@graph(
+    description="#### Owners:\n"
+    "schrockn@elementl.com, cat@elementl.com\n "
+    "#### About\n"
+    "This pipeline downloads all items from the HN API for a given day, "
+    "splits the items into stories and comment types using Spark, and uploads filtered items to "
+    "the corresponding stories or comments Snowflake table",
+)
+def download_graph():
```

Comment:
As discussed on Slack, I think a name like `download_hacker_news_actions` or something would be better because:
* More descriptive
* Avoids putting "graph" in the name.

Reply:
The main question here is do we want to just fix up the example pipeline here, or should we try to address this a different way (i.e. stricter enforcement on names)?

Reply:
Strict enforcement of names would be making the `name` argument required on `to_job`? In my opinion, it's not worth it, because it would end up requiring boilerplate in a bunch of situations where it's unnecessary.

I.e. as long as people aren't putting "graph" in the name of their graph, the naming should end up pretty clear.  We typically do not expect people to call their ops "x_op", so I think this naming convention for graphs is consistent.

Reply:
The main difference I see is that we don't expect two objects with different names in python for the op definition and the bound execution step, whereas we do have graph / job.  It may be that we decide not to require name, but I think the status quo will cause confusion.